agent/consul/internal_endpoint.go

```
@@ -69,18 +76,67 @@ func (m *Internal) NodeDump(args *structs.DCSpecificRequest,
 		&args.QueryOptions,
 		&reply.QueryMeta,
 		func(ws memdb.WatchSet, state *state.Store) error {
-			index, dump, err := state.NodeDump(ws, &args.EnterpriseMeta, args.PeerName)
-			if err != nil {
-				return err
+			peerings := make([]string, 0)
+			peerings = append(peerings, "") // this represents a local fetch
+
+			// if a PeerName is passed, then it's we only get the NodeDump for that peer
+			// otherwise, we get all the local nodes and all the nodes for every peering
+			if args.PeerName != "" {
+				peerings = append(peerings, args.PeerName)
+			} else {
+				// get a list of all peerings
+				_, listedPeerings, err := state.PeeringList(ws, args.EnterpriseMeta)
+				if err != nil {
+					m.logger.Error("could not list peers for node dump", err)
```

I think we should return the error here, rather than proceeding with peering data. Seeing that imported nodes are missing from the UI while local ones are present could cause confusion.

---

erroring out now